docs/tutorial/4-code-update-repeat.md

```diff
@@ -0,0 +1,167 @@
+> In the case of the "web" resource, it has a container image build and a Kubernetes Deployment.
+>
+> Tilt associated the `tilt-avatar-web` container image with the "web" resource because the container image name is referenced in Kubernetes YAML loaded in the `Tiltfile` with `k8s_yaml`.
+> (This is not the only way that container images can be assembled into a resource, and it's possible to manually configure where auto-assembly is insufficient.)
+
+
+## 2. Resource Update
+Now, the update process starts:
+```log
+STEP 1/3 — Building Dockerfile: [tilt-avatar-web]
+  ...
+
+STEP 2/3 — Pushing localhost:44099/tilt-avatar-web:tilt-0b9fcdf9cfea47ba
+  ...
+
+STEP 3/3 — Deploying
+     Injecting images into Kubernetes YAML
+     Applying via kubectl:
+     → web:deployment
+```
+First, Tilt built an updated version of the container image.
+Then, it pushed it to our local registry so that it can be used by Kubernetes.
+(This step could look different for you!
+Tilt adapts its workflow based on your local cluster setup, which might not require image pushes.)
+Finally, it deployed the updated image.
+
+> 🏷 **Immutable Image Tags**
+>
+> Tilt tags every image it builds with a unique `:tilt-<hash>` tag.
+> It then rewrites the Kubernetes YAML (or Helm chart) on the fly during deployment to use this tag.
+>
+> Why?
+> Using a "rolling" tag (such as `:latest`) can result in hard-to-debug issues depending on factors like image pull policy configuration.
+> With an immutable tag, you're guaranteed _exactly_ what got built is what will run.
+>
+> It's just one more thing Tilt takes care of without any extra configuration to save you a headache later!
+
+## 3. Resource Runtime Monitoring
+Once deployed, Tilt starts tracking the updated version of the resource:
+```log
+Tracking new pod rollout (web-7f9b8b65f4-wt97k):
+     ┊ Scheduled       - <1s
+     ┊ Initialized     - <1s
+     ┊ Ready           - 1s
+```
+As Tilt waits for the resource to become ready, it'll pass along relevant events, such as image pull status or container crashes, so you don't need to resort to manually investigating a failed deploy with `kubectl`.
+
+Once the container has started, Tilt will stream the logs.
+In our case, since we enabled more verbose logging for Vite (the dev server that hosts the frontend), we should see some messages as it starts up:
+```log
+yarn run v1.22.5
+$ vite
+Pre-bundling dependencies:
+  react
+  react-dom
+(this will be run only when your dependencies or config have changed)
+
+    ...
+
+  ready in 946ms.
+```
+
+If you're a bit underwhelmed by changing a log level - you caught us!
+The [Tilt Avatars][repo-tilt-avatars] project is configured to use Live Update for regular development, so we purposefully made a change in a config file that meant the full container would be rebuilt.
+
+Let's move on to the next section where we'll make more interesting code changes with Live Update.
```

Comment:
overall, i really appreciate the level of detail and explanation you're giving. it is a lot of reading, but in general, i like docs that explain how a system works because i find i'm able to better debug issues as i ramp up in a new framework/library/etc.

Reply:
Thanks! This page definitely has some longer sections like breaking down the arguments from `docker_build`, but I hope with the `<h3>`s it's skimmable to skip past something if someone isn't interested.

That said, pointing back to one of the other threads - I want to do a full "visual" pass on this once in the new design and aggressively edit text-heavy sections and/or add some more flair to break things up